### ISSUE TITLE
ValueError: Service is not complete: None - /YamahaRemoteControl/ctrl - None

### ISSUE THREAD

**Issue by rsobik** [bpannier/simpletr64#4]
I got a receiver that is connected via ethernet to my local network. When running a discovery of my devices I receive an error.

What do I have to do to make simpletr64 compatible with my receiver?

For Reference data I gathered so far. Discovery:

```
rsobik:~/ $ simpletr64_discover                                                                                                    [12:45:47]
Start discovery.
Amount of hosts found: 3
Processing: 
Host: fritz.box
Host: pascal.fritz.box
Traceback (most recent call last):
  File "/usr/local/bin/simpletr64_discover", line 108, in <module>
    box.loadDeviceDefinitions(sResult["result"].location, timeout=use_timeout)
  File "/usr/local/lib/python3.5/site-packages/simpletr64/devicetr64.py", line 648, in loadDeviceDefinitions
    self._iterateToFindSCPDElements(root, baseURIPath)
  File "/usr/local/lib/python3.5/site-packages/simpletr64/devicetr64.py", line 706, in _iterateToFindSCPDElements
    self._iterateToFindSCPDElements(child, baseURIPath)
  File "/usr/local/lib/python3.5/site-packages/simpletr64/devicetr64.py", line 661, in _iterateToFindSCPDElements
    self._processServiceList(child,baseURIPath)
  File "/usr/local/lib/python3.5/site-packages/simpletr64/devicetr64.py", line 759, in _processServiceList
    str(controlURL) + " - " + str(scpdURL))
ValueError: Service is not complete: None - /YamahaRemoteControl/ctrl - None
```

Trying to get device information I receive a similar error:

```
rsobik:~/ $ simpletr64_deviceinfo pascal.fritz.box                                                                                 [12:46:05]
Traceback (most recent call last):
  File "/usr/local/bin/simpletr64_deviceinfo", line 59, in <module>
    box.loadDeviceDefinitions(result.location, timeout=use_timeout)
  File "/usr/local/lib/python3.5/site-packages/simpletr64/devicetr64.py", line 648, in loadDeviceDefinitions
    self._iterateToFindSCPDElements(root, baseURIPath)
  File "/usr/local/lib/python3.5/site-packages/simpletr64/devicetr64.py", line 706, in _iterateToFindSCPDElements
    self._iterateToFindSCPDElements(child, baseURIPath)
  File "/usr/local/lib/python3.5/site-packages/simpletr64/devicetr64.py", line 661, in _iterateToFindSCPDElements
    self._processServiceList(child,baseURIPath)
  File "/usr/local/lib/python3.5/site-packages/simpletr64/devicetr64.py", line 759, in _processServiceList
    str(controlURL) + " - " + str(scpdURL))
ValueError: Service is not complete: None - /YamahaRemoteControl/ctrl - None
```

`http://pascal.fritz.box:8080/MediaRenderer/desc.xml` has the following content:

``` xml
<?xml version="1.0"?>
<root
  xmlns="urn:schemas-upnp-org:device-1-0"
  xmlns:ms="urn:microsoft-com:wmc-1-0"
  xmlns:pnpx="http://schemas.microsoft.com/windows/pnpx/2005/11"
  xmlns:df="http://schemas.microsoft.com/windows/2008/09/devicefoundation"
  xmlns:yamaha="urn:schemas-yamaha-com:device-1-0">
    <yamaha:X_device><yamaha:X_URLBase>http://REDACTED:80/</yamaha:X_URLBase><yamaha:X_serviceList><yamaha:X_service><yamaha:X_specType>urn:schemas-yamaha-com:service:X_YamahaRemoteControl:1</yamaha:X_specType><yamaha:X_controlURL>/YamahaRemoteControl/ctrl</yamaha:X_controlURL><yamaha:X_unitDescURL>/YamahaRemoteControl/desc.xml</yamaha:X_unitDescURL></yamaha:X_service></yamaha:X_serviceList></yamaha:X_device>
    <specVersion>
        <major>1</major>
        <minor>0</minor>
    </specVersion>
    <device
      ms:X_MS_SupportsWMDRM="true">
        <dlna:X_DLNADOC xmlns:dlna="urn:schemas-dlna-org:device-1-0">DMR-1.50</dlna:X_DLNADOC>
        <pnpx:X_compatibleId>MS_DigitalMediaDeviceClass_DMR_V001
                </pnpx:X_compatibleId>
        <pnpx:X_deviceCategory>MediaDevices Multimedia.DMR MediaDevice.DMC
                </pnpx:X_deviceCategory>
        <pnpx:X_hardwareId>VEN_0033&amp;DEV_0006&amp;REV_01
                </pnpx:X_hardwareId>
        <df:X_deviceCategory>Multimedia.DMR
                </df:X_deviceCategory>
        <deviceType>urn:schemas-upnp-org:device:MediaRenderer:1</deviceType>
        <friendlyName>Pascal</friendlyName>
        <manufacturer>Yamaha Corporation</manufacturer>
        <manufacturerURL>http://www.yamaha.com/</manufacturerURL>
        <modelDescription>AV Receiver</modelDescription>
        <modelName>RX-V475</modelName>
        <modelNumber>V475</modelNumber>
        <modelURL>http://www.yamaha.com/</modelURL>
        <serialNumber>REDACTED</serialNumber>
        <UDN>REDACTED</UDN>
        <UPC>REDACTED</UPC>
        <iconList>
            <icon>
                <mimetype>image/jpeg</mimetype>
                <width>48</width>
                <height>48</height>
                <depth>24</depth>
                <url>/BCO_device_sm_icon.jpg</url>
            </icon>
            <icon>
                <mimetype>image/jpeg</mimetype>
                <width>120</width>
                <height>120</height>
                <depth>24</depth>
                <url>/BCO_device_lrg_icon.jpg</url>
            </icon>
            <icon>
                <mimetype>image/png</mimetype>
                <width>48</width>
                <height>48</height>
                <depth>24</depth>
                <url>/BCO_device_sm_icon.png</url>
            </icon>
            <icon>
                <mimetype>image/png</mimetype>
                <width>120</width>
                <height>120</height>
                <depth>24</depth>
                <url>/BCO_device_lrg_icon.png</url>
            </icon>
        </iconList>
        <serviceList>
            <service>
                <serviceType>urn:schemas-upnp-org:service:RenderingControl:1</serviceType>
                <serviceId>urn:upnp-org:serviceId:RenderingControl</serviceId>
                <SCPDURL>/RenderingControl/desc.xml</SCPDURL>
                <controlURL>/RenderingControl/ctrl</controlURL>
                <eventSubURL>/RenderingControl/evt</eventSubURL>
            </service>
            <service>
                <serviceType>urn:schemas-upnp-org:service:ConnectionManager:1</serviceType>
                <serviceId>urn:upnp-org:serviceId:ConnectionManager</serviceId>
                <SCPDURL>/ConnectionManager/desc.xml</SCPDURL>
                <controlURL>/ConnectionManager/ctrl</controlURL>
                <eventSubURL>/ConnectionManager/evt</eventSubURL>
            </service>
            <service>
                <serviceType>urn:schemas-upnp-org:service:AVTransport:1</serviceType>
                <serviceId>urn:upnp-org:serviceId:AVTransport</serviceId>
                <SCPDURL>/AVTransport/desc.xml</SCPDURL>
                <controlURL>/AVTransport/ctrl</controlURL>
                <eventSubURL>/AVTransport/evt</eventSubURL>
            </service>
        </serviceList>
        <presentationURL>http://REDACTED/</presentationURL>
    </device>
</root>
```

**Comment by bpannier** [comment]
Please, update to latest release. Does it work now?

**Comment by bpannier** [comment]
Resolved in 1.0.5
